### PR TITLE
build(file-driver): use fluid-tsc

### PR DESCRIPTION
**Pull request by DLehenbauer**
Semi-automatic PR to switch to fluid-tsc

### CODE REVIEW

File: packages/drivers/file-driver/package.json [hunk 1]
```diff
@@ -11,16 +11,50 @@
 	"license": "MIT",
 	"author": "Microsoft and contributors",
 	"sideEffects": false,
-	"type": "commonjs",
+	"type": "module",
+	"exports": {
+		".": {
+			"import": {
+				"types": "./lib/index.d.ts",
+				"default": "./lib/index.js"
```
Comment (RishhiB) [inline packages/drivers/file-driver/package.json:19]: Are we trimming these in a separate PR?

File: packages/drivers/file-driver/package.json [hunk 2]
```diff
@@ -64,13 +98,14 @@
...
+					"api-extractor:commonjs",
+					"api-extractor:esnext"
 				],
 				"script": false
 			},
 			"tsc": [
-				"...",
-				"typetests:gen"
+				"typetests:gen",
+				"^tsc"
```
Comment (RishhiB) [inline packages/drivers/file-driver/package.json:108]: for my own knowledge. why did we have to add this?

**Review by RishhiB — APPROVED** [review]

File: packages/drivers/file-driver/package.json (hunk 1 above)
Comment (DLehenbauer) [reply packages/drivers/file-driver/package.json:19]: Yeah.  When we trim, we have to convert all downstream dependencies to node16, so we're doing that in a separate step.

File: packages/drivers/file-driver/package.json (hunk 2 above)
Comment (DLehenbauer) [reply packages/drivers/file-driver/package.json]: That is an error introduce by 'policy-check:fix'.  I've put it back.  Thank you for noticing!

**Pull request merged**